src/Optimization/UI/AxisymmetricOptimization.jl

```diff
@@ -0,0 +1,349 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+module AxisymmetricOptimization
+
+using OpticSim, OpticSim.Geometry, OpticSim.Emitters
+using CSV, DataFrames
+using Ipopt
+using JuMP
+import DataStructures
+
+global global_model = nothing
+global global_ax = nothing
+global global_df = nothing
+global global_counter = 0
+global solution_history = []
+global statistics = Dict()
+global options = Dict()
+
+function set_global_ax(ax) 
+    global global_ax = ax
+end 
+
+global emitter = Sources.Source(origins=Emitters.Origins.Hexapolar(5, 8.0, 8.0), directions=Directions.Constant(), transform=Transform(Vec3(0.0,0.0,5.0), Geometry.unitZ3()*-1))
+
+mutable struct VarInfo
+    _row::Int
+    _col::Int
+    _name::String
+    _start_value::Float64
+    _lower_bound::Union{Float64,Nothing}
+    _upper_bound::Union{Float64,Nothing}
+    _info::Dict
+end
+
+function get_number(dic, key)
+    v = get(dic, key, Nothing)
+    if (v != Nothing)
```

Comment:
n.b. `Nothing` is a type - `nothing` is the actual variable that should be used

it's also more "Julian" to use `isnothing` as this makes use of type dispatch without having to do any comparisons

Reply:
Julia Computing tech support told us that we should use

``` v === nothing```
and 
```v !== nothing```

instead of
```isnothing(v)```

They said the compiler can do more extensive optimization with the === and !== statements.

Reply:
Interesting, thanks for the correction. I was looking at this link and didn't read all the way to the end. I'll have to correct some of my code then. https://stackoverflow.com/questions/56852880/comparing-julia-variable-to-nothing-using-or